votrfront/js/PrehladStudiaPage.js

```diff
@@ -81,6 +83,23 @@ export function PrehladStudiaStudia() {
       studia, PrehladStudiumColumns, query, 'studiaSort');
 
     var message = studia.length ? null : "V AISe nemáte žiadne štúdiá.";
+    
+    var button = function (studium) {
+        var zapisne_listy = cache.get('get_zapisne_listy', studium.studium_key);
+        var aktualny_zapisny_list = 0;
+        if (zapisne_listy !== null){
+            aktualny_zapisny_list = zapisne_listy.filter(zl => zl.akademicky_rok === currentAcademicYear()).length;
+        }
+        if (studium.koniec === '') { // ak este neskoncilo studium
+            if (aktualny_zapisny_list !== 0) { // ak uz mame zapisny list na tento rok
+                return <button class='btn btn-xs btn-success appear-disabled'>Pridať zápisný list</button>
+            } else { //  ak studium prebieha a nemame este zapisny list na tento rok
+                return <button onClick={() => sendRpc('create_zapisny_list', [studium.studium_key, currentAcademicYear()], (message)=>{if (message !== null) {alert(message);} else {RequestCache.invalidate('get_zapisne_listy');}})} class='btn btn-xs btn-success'>Pridať zápisný list</button>
```

Comment:
Rozdel to na viac riadkov.

Reply:
pokusila som sa to rozdelit, ale neviem ci dobre